engine/gameobject/src/gameobject/res_collection.cpp

```diff
@@ -252,45 +253,45 @@ namespace dmGameObject
         return size;
     }
 
-    static dmResource::Result ResCollectionCreate(const dmResource::ResourceCreateParams& params)
+    static dmResource::Result ResCollectionCreate(const dmResource::ResourceCreateParams* params)
     {
-        Register* regist = (Register*) params.m_Context;
-        dmGameObjectDDF::CollectionDesc* collection_desc = (dmGameObjectDDF::CollectionDesc*) params.m_PreloadData;
+        Register* regist = (Register*) params->m_Context;
+        dmGameObjectDDF::CollectionDesc* collection_desc = (dmGameObjectDDF::CollectionDesc*) params->m_PreloadData;
 
         HCollection hcollection = 0;
-        dmResource::Result res = AcquireResources(collection_desc->m_Name, params.m_Factory, regist, collection_desc, params.m_Filename, &hcollection);
+        dmResource::Result res = AcquireResources(collection_desc->m_Name, params->m_Factory, regist, collection_desc, params->m_Filename, &hcollection);
         dmDDF::FreeMessage(collection_desc);
 
         if (res != dmResource::RESULT_OK)
         {
             return res;
         }
 
-        params.m_Resource->m_Resource = hcollection;
-        params.m_Resource->m_ResourceSize = CalcSize(hcollection->m_Collection);
+        ResourceDescriptorSetResource(params->m_Resource, hcollection);
+        ResourceDescriptorSetResourceSize(params->m_Resource, CalcSize(hcollection->m_Collection));
```

Comment:
... I did however move the resource descriptor and type into new handle types, to keep the internals hidden.